libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialog.cs

```diff
@@ -562,14 +584,19 @@ private async Task<bool> QueueFirstMatchAsync(SequenceContext sequenceContext, D
             return false;
         }
 
-        private void EnsureDependenciesInstalled()
+        private void EnsureDependenciesInstalled(DialogContext dc)
         {
             lock (this)
             {
                 if (!installedDependencies)
                 {
                     installedDependencies = true;
 
+                    var needsTracker = false;
```

Comment:
This code is inside a "lock(this)", which is considered an anti-pattern. Consider refactoring. 
https://jonskeet.uk/csharp/threads/lockchoice.html
https://stackoverflow.com/questions/251391/why-is-lockthis-bad